samples/gallery/src/main/java/com/bumptech/glide/samples/gallery/HorizontalGalleryFragment.kt

```diff
@@ -1,50 +1,559 @@
+   * @param numberOfItemsToPreload The number of items to preload ahead of the user's current
+   * position. This should be tested for each application. If the total memory size of the preloaded
+   * images exceeds the memory cache, preloading is effectively useless. However if you preload too
+   * few things, the buffer may be small enough that images are not available when they could be, so
+   * it's always a balancing act. The smaller the preloaded image, the more you can preload.
+   * @param fixedVisibleItemCount The number of visible items. In some cases this can vary widely
+   * in which case you can leave this value `null`. If the number of visible items is always one
+   * or two, it might make sense to just set this to the larger of the two to reduce churn in the
+   * preloader.
+   * @param requestBuilderModifier See [ListPreloader.PreloadModelProvider.getPreloadRequestBuilder]
+   */
+  // TODO(judds): Consider wrapping a LazyRow / LazyColumn and providing state instead of a separate
+  // function. Wrapping might also make it easier to pass through the size and request builder
+  // modifications so that it's easier to make sure the preload size matches a size on the
+  // GlideImage
+  @Composable
+  fun <DataTypeT : Any> GlideLazyListPreloader(
+    state: LazyListState,
+    data: List<DataTypeT>,
+    size: Size,
+    numberOfItemsToPreload: Int,
+    fixedVisibleItemCount: Int? = null,
+    requestBuilderModifier:
+      (data: DataTypeT, requestBuilder: RequestBuilder<*>) -> RequestBuilder<*>,
+  ) {
+    Log.d("TEST", "GlidePreloader")
+    val preloader =
+      rememberGlidePreloader(
+        data = data,
+        size = size,
+        numberOfItemsToPreload = numberOfItemsToPreload,
+        requestBuilderModifier = requestBuilderModifier
+      )
+    doPreload(preloader = preloader, state = state, fixedVisibleItemCount = fixedVisibleItemCount)
+  }
+
+  @Composable
+  private fun <DataTypeT: Any> doPreload(
+    preloader: ListPreloader<DataTypeT>, state: LazyListState, fixedVisibleItemCount: Int?
+  ) {
+    Log.d("TEST", "doPreload")
+    val lazyListVisibleInfo = derivedStateOf { state.lazyListVisibleInfo(fixedVisibleItemCount) }
+    // TODO(judds): Without this remember call, we're never recomposed. It's unclear what the right
+    // way to do this is?
+    remember(lazyListVisibleInfo.value) {}
+    SideEffect {
+      preloader.onScroll(
+        null,
+        lazyListVisibleInfo.value.firstVisibleItemIndex,
+        lazyListVisibleInfo.value.visibleItemCount,
+        lazyListVisibleInfo.value.totalItemCount)
+    }
+  }
+
+  @Composable
+  private fun <DataTypeT: Any> rememberGlidePreloader(
+    data: List<DataTypeT>,
+    size: Size,
+    numberOfItemsToPreload: Int,
+    requestBuilderModifier:
+      (data: DataTypeT, requestBuilder: RequestBuilder<*>) -> RequestBuilder<*>,
+  ): ListPreloader<DataTypeT> {
+    Log.d("TEST", "rememberGlidePreloader")
+    val currentData = remember(data) { data }
```

Comment:
This is redundant, just read `data` directly. This actually introduces overhead, since `remember` has to read from and write to the slot table, which isn't free.

Reply:
Done